### ISSUE TITLE
Auto commit doesn't work with kafka 0.9-0.10.0.1 when handlers spend too much time

### ISSUE THREAD

**Issue by Frank-Hust** [spring-projects/spring-kafka#273]
such versions of kafka consumer clients use poll method in a sync and blocking way to send heartbeat to broker,  not in background thread. 
If set `spring.kafka.consumer.enable-auto-commit=true` and handlers use too much time more than session timeout, like the code below, then the poll method won't be executed again before the handlers complete, as a result , the heartbeat won't be sent  in time, which causes partition revocation, and reassignation, and the record will be reconsumer again and again
```java
     @KafkaListener(topics = "newTopic")
    public void processMessage(String content, @Header(KafkaHeaders.RECEIVED_PARTITION_ID) int partition) throws InterruptedException {
        logger.info("start to sleep: partition {} content: {}", partition, content);
        System.out.println();
        Thread.sleep(100_000);
        logger.info("resume from sleep: partition {} content: {}", partition, content);
    }
```
The input look like this
```
2017-04-05 20:19:08.274  INFO 7476 --- [afka-consumer-1] com.demo.MyBean                          : start to sleep: partition 1 content: myself

2017-04-05 20:19:39.544  INFO 7476 --- [afka-consumer-1] o.a.k.c.c.internals.ConsumerCoordinator  : Setting newly assigned partitions [newTopic-1, newTopic-2, newTopic-3, newTopic-0] for group myGroup

2017-04-05 20:19:39.567  INFO 7476 --- [afka-consumer-1] com.demo.MyBean                          : start to sleep: partition 1 content: myself
```

**Comment by garyrussell** [comment]
Right; that is a known limitation. When auto commit is false, we hand off the messages to another thread so we can keep polling while the slow consumer is doing its work; if we detect a slow consumer we `pause()` the consumer so we can keep polling, but not get any messages and `resume()` when the listener catches up.

Unfortunately, with auto commit true, we can't do that, because if we did, the commit would happen immediately after the handoff to the listener thread, which would result in lost messages.

Trust me, it was a pain to implement the second thread handoff and I am glad they finally fixed the heartbeat issue so we can rip all that out at some point.

The bottom line is auto commit with a slow consumer will not work with those client versions.

**Comment by artembilan** [comment]
> The bottom line is auto commit with a slow consumer will not work with those client versions.

That even true from just a raw Apache Kafka client. Not a Spring Kafka issue, in other words.
